test/core/sector_storage/sector_storage_test.cpp

```diff
@@ -0,0 +1,274 @@
+using fc::proofs::PieceData;
+using fc::proofs::PieceInfo;
+using fc::proofs::UnpaddedPieceSize;
+using fc::sector_storage::InteractiveRandomness;
+using fc::sector_storage::SectorStorage;
+using fc::sector_storage::SectorStorageError;
+using fc::sector_storage::SectorStorageImpl;
+using fc::storage::filestore::Path;
+using proofs = fc::proofs::Proofs;
+
+class SectorStorageTest : public test::BaseFS_Test {
+ public:
+  SectorStorageTest() : test::BaseFS_Test("fc_sector_storage_test") {
+    sector_storage_ = std::make_unique<SectorStorageImpl>(
+        base_path.string(), post_proof_, seal_proof_);
+  }
+
+ protected:
+  RegisteredProof seal_proof_ = RegisteredProof::StackedDRG2KiBSeal;
+  RegisteredProof post_proof_ = RegisteredProof::StackedDRG2KiBPoSt;
+  std::unique_ptr<SectorStorage> sector_storage_;
+};
+
+/**
+ * @given sector
+ * @when want to get path to sealed sector
+ * @then sealed sector was obtained. cache and unsealed paths are empty
+ */
+TEST_F(SectorStorageTest, AcquireSector_Sealed) {
+  std::string sealed_result =
+      (base_path / fs::path("sealed") / fs::path("s-t01-3")).string();
+
+  SectorId sector{
+      .miner = 1,
+      .sector = 3,
+  };
+
+  EXPECT_OUTCOME_TRUE(paths,
+                      sector_storage_->acquireSector(
+                          sector, SectorFileTypes::FTSealed, 0, true))
+
+  ASSERT_TRUE(paths.cache.empty());
+  ASSERT_TRUE(paths.unsealed.empty());
+  ASSERT_EQ(paths.sealed, sealed_result);
+}
+
+/**
+ * @given sector
+ * @when want to get path to unsealed sector
+ * @then unsealed sector was obtained. cache and sealed paths are empty
+ */
+TEST_F(SectorStorageTest, AcquireSector_Unsealed) {
+  std::string cache_result = "";
+  std::string unsealed_result =
+      (base_path / fs::path("unsealed") / fs::path("s-t01-3")).string();
+  std::string sealed_result = "";
+
+  SectorId sector{
+      .miner = 1,
+      .sector = 3,
+  };
+
+  EXPECT_OUTCOME_TRUE(paths,
+                      sector_storage_->acquireSector(
+                          sector, SectorFileTypes::FTUnsealed, 0, true))
+
+  ASSERT_TRUE(paths.cache.empty());
+  ASSERT_EQ(paths.unsealed, unsealed_result);
+  ASSERT_TRUE(paths.sealed.empty());
+}
+
+/**
+ * @given sector
+ * @when want to get path to cache sector
+ * @then all path were obtained
+ */
+TEST_F(SectorStorageTest, AcquireSector_Cache) {
+  std::string cache_result =
+      (base_path / fs::path("cache") / fs::path("s-t01-3")).string();
+  std::string unsealed_result =
+      (base_path / fs::path("unsealed") / fs::path("s-t01-3")).string();
+  std::string sealed_result =
+      (base_path / fs::path("sealed") / fs::path("s-t01-3")).string();
+
+  SectorId sector{
+      .miner = 1,
+      .sector = 3,
+  };
+
+  EXPECT_OUTCOME_TRUE(
+      paths,
+      sector_storage_->acquireSector(sector, SectorFileTypes::FTCache, 0, true))
+
+  ASSERT_EQ(paths.cache, cache_result);
+  ASSERT_EQ(paths.unsealed, unsealed_result);
+  ASSERT_EQ(paths.sealed, sealed_result);
```

Comment:
Wrong test, "cache" is disjoint from "sealed" and "unsealed"

Reply:
Fixed